Add in ADR's (#127)
Western-Health-Covid19-Collaboration/wh_covid19_app: docs/Architecture/ADR-0002 Embed content in the app.md

# Embed content in the app

## Status

Accepted

## Context

> The app primarily displays information for medical staff working in hospitals.  The information needs to come from somewhere and there are a few options:

Reviewer: ```suggestion
The app primarily displays information for medical staff working in hospitals.  The information and data need to come from somewhere, and there are a few options:
```

### Download content from a CMS/S3 bucket/Website

> We could pull the information in the app from somewhere on the web.  Some kind of CMS, S3 bucket or website where the files are hosted would be the most obvious choices.

Reviewer: ```suggestion
We could pull the information in the app from somewhere on the web.  Some of the more obvious choices, where the files are stored, include CMS, S3 bucket or website.
```

> This does make it easier to quickly the information in the app, but would require the build out of more infrastructure and architecture to support all this.

Reviewer: ```suggestion
With the suggestion aforementioned, it will be more accessible and quicker to collect the information in the app. However, it would require the build-out of more infrastructure and architecture to support all this.
```

> It also could mean that if the internet connection goes down to a hospital in  a crisis any information that isn't already cached in the app would be unavailable to users.

Reviewer: ```suggestion
If there is an internet failure connection, where it leads to any information that was not cached before the outage, then the contents and information of the app would be unavailable to users.
```

### Embed the content in the app

> An alternative is to embed all the information in the app, so its all included when the installs the app.

Reviewer: ```suggestion
The alternative method is to embed all of the information in the app so that all of the information is included, after downloading the app.
```

This is architecturally much simpler and means all the content will be available offline if the internet should go down.

> It does mean that to update the information a new version of the app needs to be released.

Reviewer: ```suggestion
Subsequently, a new version of the app needs to be released to update the information.
```

## Decision

For the initial versions of the application, we will embed the content in the app.

> Later once the situation stablises and we need to support more hospitals, we  can re-visit the decision and decide if we want to download content.

Reviewer: ```suggestion
Once the situation stabilises, and we can provide support to more hospitals, we can re-visit the decision and decide if we want to download content.
```
